### ISSUE TITLE
Add support for extra fields while propagating in SpanContext

### ISSUE THREAD

**Issue by MrLina** [openzipkin/zipkin-go#112]
while propagating，it support fields like TraceId、SpanId、ParentSpanId 、Debug And Sample。But No extra fields to propagate other user-defined fields 。The feature has been supported In zipkin-Java
 [ExtraFieldPropagation](https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/propagation/ExtraFieldPropagation.java)  [TraceContext](https://sourcegraph.com/github.com/openzipkin/brave@master/-/blob/brave/src/main/java/brave/propagation/TraceContext.java#L173:26&tab=def)
Hope the feature can also be supported In zipkin-Go

**Comment by jcchavezs** [comment]
This is something definitively desirable by itself but before we start
working on something like this I would like to hear more about your use
case. Could you please give us more details?

**Comment by MrLina** [comment]
We need to propagating some mark in all the Trace link,for example ,we need to do full link pressure measurement，so we need propagating the mark of pressure measurement in all the Trace link to divided the normal reques and the test request。 And 
we also need to propagating other mark such as "userId" 、"orderId" in all the Trace link in order to find the request by the "userId" 。And we can know the different request such as normal request or error request or test request or others by special mark， if some mark can be propagating the all the Trace link，then we can do many thing。

**Comment by jcchavezs** [comment]
I see. If the purpose is filtering, if you just tag the root span (or any
span) with the desired mark then in zipkin UI you can use that filtering.
Does that solve your problem?

Den man. 4. mar. 2019, 05:06 skrev MrLina <notifications@github.com>:

**Comment by MrLina** [comment]
No, Our purpose is not only tag the span .The tag will repoter to server ,and we can search the span by the tag ,but it's  can not be propagating in all the Trace link,   We hope propagate the user-defined mark in the all Trace link, then we can  get the special mark to do other thing from any invoke which the application is instumented .it's not only filter from zipkin UI. In my case，such as  pressure measurement ，we need propagated the pressure mark in all the Trace link。Then other application get the special pressure mark from the Trace link，they will do other thing by the mark。
I think propagating the extra fields is really a good feature。Hope you can support。
Thank you for your reply。

**Comment by basvanbeek** [comment]
I'm in favor of having a brave like explicit white list of additional fields to propagate. I'm open to accepting PR's for this.
